### ISSUE TITLE
Unable to create SQS redrive policy using CloudFormation

### ISSUE THREAD

**Issue by osvaldolove** [localstack/localstack#1416]
Hi,

I'm using the latest localstack docker image which is currently `0.9.5`.

The following error occurs when trying to use CloudFormation to create Dead Letter Queue and Queue in same template.
`An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the CreateStack operation: Could not find DLQ for arn:aws:sqs:elasticmq:000000000000:test-dlq`

Example template
```yaml
AWSTemplateFormatVersion: 2010-09-09
Resources:
  DeadLetterQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: 'test-dlq'
      DelaySeconds: 0
      MaximumMessageSize: 262144 # 256KB
      MessageRetentionPeriod: 1209600 # 14 Days
      VisibilityTimeout: 120 # 2 minutes
  Queue:
    Type: AWS::SQS::Queue
    DependsOn: DeadLetterQueue
    Properties:
      QueueName: 'test-queue'
      DelaySeconds: 0
      MaximumMessageSize: 262144 # 256KB
      MessageRetentionPeriod: 345600 # 4 Days
      RedrivePolicy:
        deadLetterTargetArn:
          Fn::GetAtt:
            - DeadLetterQueue
            - Arn
        maxReceiveCount: 1
      VisibilityTimeout: 120 # 2 minutes
```

Note, I even tried to use the `DependsOn` property to force it to create the DLQ first but to no avail.

Command used with `awslocal`...
```shell
awslocal cloudformation create-stack --stack-name test1 --template-body file://./queues.template.dlq.yml
```
Using the same template in AWS CloudFormation works as expected.

*Debug log output*
```
localstack_1  | 2019-07-10T10:35:16:DEBUG:__main__: Currently updating stack resource test1/DeadLetterQueue: None 
localstack_1  | 2019-07-10T10:35:16:DEBUG:__main__: Deploying CloudFormation resource: {'Type': 'AWS::SQS::Queue', 'Properties': {'QueueName': 'test-dlq', 'DelaySeconds': 0, 'MaximumMessageSize': 262144, 'MessageRetentionPeriod': 1209600, 'VisibilityTimeout': 120}}
localstack_1  | 2019-07-10T10:35:16:DEBUG:localstack.utils.cloudformation.template_deployer: Deploying resource type "SQS::Queue" id "DeadLetterQueue"
localstack_1  | 2019-07-10T10:35:16:DEBUG:localstack.utils.cloudformation.template_deployer: Request for creating resource type "SQS::Queue": {'QueueName': 'test-dlq'}
localstack_1  | 10:35:16.440 [elasticmq-akka.actor.default-dispatcher-3] INFO  o.elasticmq.actor.QueueManagerActor - Creating queue QueueData(test-dlq,MillisVisibilityTimeout(30000),PT0S,PT0S,2019-07-10T10:35:16.408Z,2019-07-10T10:35:16.408Z,None,false,false,None,None,Map()) 
localstack_1  | 2019-07-10T10:35:16:DEBUG:__main__: Currently updating stack resource test1/Queue: None
localstack_1  | 2019-07-10T10:35:16:ERROR:__main__: Unable to parse and create resource "Queue": 400 Bad Request: <?xml version="1.0" encoding="UTF-8"?> 
localstack_1  | <Response>
localstack_1  | <Errors>
localstack_1  | <Error> 
localstack_1  | <Code>AWS.SimpleQueueService.NonExistentQueue</Code>
localstack_1  | <Message>Could not find DLQ for arn:aws:sqs:elasticmq:000000000000:test-dlq</Message>
localstack_1  |
localstack_1  | </Error>
localstack_1  | </Errors>
localstack_1  | <RequestID>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</RequestID> 
localstack_1  | </Response>
localstack_1  | 2019-07-10T10:35:16:INFO:werkzeug: 127.0.0.1 - - [10/Jul/2019 10:35:16] "POST / HTTP/1.1" 400 -
localstack_1  | 2019-07-10T10:35:16:DEBUG:localstack.services.cloudformation.cloudformation_listener: Error response from CloudFormation (400) POST /: b'<?xml version="1.0" encoding="UTF-8"?>\n  <Response>\n    <Errors>\n      <Error>\n        <Code>AWS.SimpleQueueService.NonExistentQueue</Code>\n        <Message>Could not find DLQ for arn:aws:sqs:elasticmq:000000000000:test-dlq</Message>\n        \n      </Error>\n    </Errors>\n  <RequestID>7a62c49f-347e-4fc4-9331-6e8eEXAMPLE</RequestID>\n</Response>'
```

Can you help? Thanks.

**Comment by robertodias** [comment]
I am facing exactly the same issue, running on MacOS with fish and virtualenv. Tried to force elasticMQ downgrade but with no sucess.

**Comment by maxenglander** [comment]
I did some digging into this, and added some debugging statements into `moto`:

In `moto/sqs/models.py#266`:
```
        f.write("looking for a queue matching %s\n" % (self.redrive_policy['deadLetterTargetArn']))
        for queue in sqs_backends[self.region].queues.values():
            f.write("found a queue: %s\n" % (queue.queue_arn))
            if queue.queue_arn == self.redrive_policy['deadLetterTargetArn']:
```

And the result:
```
$ cat /tmp/moto.log
looking for a queue matching arn:aws:sqs:us-east-1:000000000000:dlq
found a queue: arn:aws:sqs:us-east-1:123456789012:dlq
```

Notice the mismatched account IDs.

**Comment by maxenglander** [comment]
Did some more digging. 

My specific use case is that I am using the Serverless Framework to define a dead letter queue (DLQ) and another queue that uses that DLQ in its redrive policy. I believe what is happening is that the Serverless Framework creates and uploads a CloudFormation template that contains a redrive policy that looks like this:

```
RedrivePolicy:
  deadLetterTargetArn:
    Fn::GetAtt:
      - dlq
      - Arn
```

Localstack forwards a CFN deployment request to Moto, which then downloads the Serverless-generated CFN template from S3. As this template is processed, and resources are created, `Fn::GetAtt[dlq, Arn]` is eventually evaluated by Localstack, and Localstack returns the ARN, which contains the account ID `000000000000` baked into it. This Arn is eventually passed to Moto in an `sqs:CreateQueue` call. Moto then compares this ARN against queues that it is aware of internally, and fails to find any match.

Since Localstack and Moto seem to call back and forth to each other, I might not have some of these interactions right.

**Comment by whummer** [comment]
Thanks all. This should be fixed in #1542 - please re-open this issue if the problem persists.
